mobile/src/utils/index.js

```diff
@@ -36,13 +36,19 @@ export function evenlySplitAddress(address = '') {
 export function truncateAddress(address = '', chars = 5) {
   const MAX_ADDRESS_LENGTH = 10
 
-  if (address.length <= MAX_ADDRESS_LENGTH) return address
+  if (
+    address.length <= MAX_ADDRESS_LENGTH ||
+    chars === -1 ||
+    chars > address.length
+  )
+    return address
 
   const separator = '...'
+  const firstPortionLength = Math.floor(chars / 2)
   return (
-    address.substr(0, chars) +
+    address.substr(0, firstPortionLength) +
     separator +
-    address.substr(address.length - chars)
+    address.substr(address.length - Math.max(1, chars - 3 - firstPortionLength))
```

Comment:
nitpick: I know I suggested this part, taking a second look perhaps `separator.length` is more explanatory than `3`

Reply:
i think that it should be :
address.substr(address.length - Math.max(3, chars - 3 - firstPortionLength))

is it correct ?

Reply:
Yeah I think you are right. Also for clarity I would upgrade it to
```
address.substr(address.length - Math.max(3, chars - separator.length - firstPortionLength))
```
to make a better sense of where the "magic 3" is coming from :)